R-package/R/lightgbm.R

```diff
@@ -25,21 +24,23 @@ NULL
 #' @param save_name File name to use when writing the trained model to disk. Should end in ".model".
 #' @param ... Additional arguments passed to \code{\link{lgb.train}}. For example
 #'     \itemize{
-#'        \item{valids}{a list of \code{lgb.Dataset} objects, used for validation}
-#'        \item{obj}{objective function, can be character or custom objective function. Examples include
+#'        \item{\code{valids}: a list of \code{lgb.Dataset} objects, used for validation}
+#'        \item{\code{obj}: objective function, can be character or custom objective function. Examples include
 #'                   \code{regression}, \code{regression_l1}, \code{huber},
 #'                    \code{binary}, \code{lambdarank}, \code{multiclass}, \code{multiclass}}
-#'        \item{eval}{evaluation function, can be (a list of) character or custom eval function}
-#'        \item{record}{Boolean, TRUE will record iteration message to \code{booster$record_evals}}
-#'        \item{colnames}{feature names, if not null, will use this to overwrite the names in dataset}
-#'        \item{categorical_feature}{list of str or int. type int represents index, type str represents feature names}
-#'        \item{reset_data}{Boolean, setting it to TRUE (not the default value) will transform the booster model
+#'        \item{\code{eval}: evaluation function, can be (a list of) character or custom eval function}
+#'        \item{\code{record}: Boolean, TRUE will record iteration message to \code{booster$record_evals}}
+#'        \item{\code{colnames}: feature names, if not null, will use this to overwrite the names in dataset}
+#'        \item{\code{categorical_feature}: categorical features. This can either be a character vector of feature
+#'                            names or an integer vector with the indices of the features (e.g. \code{c(1L, 10L)} to
+#'                            say "the first and tenth columns").}
+#'        \item{\code{reset_data}: Boolean, setting it to TRUE (not the default value) will transform the booster model
 #'                          into a predictor model which frees up memory and the original datasets}
-#'         \item{boosting}{Boosting type. \code{"gbdt"} or \code{"dart"}}
-#'         \item{num_leaves}{number of leaves in one tree. defaults to 127}
-#'         \item{max_depth}{Limit the max depth for tree model. This is used to deal with
+#'         \item{\code{boosting}: Boosting type. \code{"gbdt"} or \code{"dart"}}
+#'         \item{\code{num_leaves}: number of leaves in one tree. defaults to 127}
```

Comment:
From where this default value is taken from? If from cpp side, then it is not true anymore: https://lightgbm.readthedocs.io/en/latest/Parameters.html#num_leaves
```suggestion
#'         \item{\code{num_leaves}: Number of leaves in one tree. Defaults to 127.}
```

Reply:
😬 I'll  update it, thank you. Not something that is set on the R side.

Reply:
This line was duplicated in `lightgbm()`,  `lgb.train()` and `lgb.cv()`. I updated all three to:

```
Maximum number of leaves in one tree.
```

Reply:
Cool! Thanks!